### PR TITLE
SITCOM-1737: Fix axis error when only a single point is returned in data

### PR DESCRIPTION
**Pull request by ugyballoons**

### CODE REVIEW

File: lib/src/models/axes/mapping.dart [hunk 1]
```diff
@@ -173,6 +173,11 @@ class LinearMapping extends Mapping {
     double min = bounds.min.toDouble();
     double max = bounds.max.toDouble();
 
+    // What if max == min (e.g. there's only one point)
+    if (max == min) {
+      max += 1;
+      min -= 1;
+    }
```
Comment (fred3m) [inline lib/src/models/axes/mapping.dart:179]: I would change this to `max += pow(10, log10(max))` and the same for min, in order to correctly show the scale of the point.

File: lib/src/models/axes/mapping.dart [hunk 2]
```diff
@@ -243,6 +248,11 @@ AxisTicks getLogTicks({
 }) {
   double min = bounds.min.toDouble();
   double max = bounds.max.toDouble();
+
+  if (max == min) {
+    max += 1;
+    min -= 1;
+  }
```
Comment (fred3m) [inline lib/src/models/axes/mapping.dart:253]: Same fix here as the previous comment.

**Review by fred3m — COMMENTED** [review]
Thanks for adding a test!

File: lib/src/models/axes/mapping.dart (hunk 1 above)
Comment (ugyballoons) [reply lib/src/models/axes/mapping.dart:179]: Fantastic!

Comment (ugyballoons) [reply lib/src/models/axes/mapping.dart:179]: Although, how is this different from just `max += max`?

Comment (fred3m) [reply lib/src/models/axes/mapping.dart:179]: Yeah, I think I was still half asleep when I wrote that. I haven't tested it, but I was thinking something more like
```
power = log10(max)
min = pow(10, floor(power)) * 0.8  // Add 20% padding
max = pow(10, ceil(power)) * 1.2
```
which would have nice logarithmic bounds. But now that I think about it we probably want the lone data point in the center, so something like
```
radius = max/2
min -= radius
max += radius
```
This should give us bounds that have a range of the value itself, with the value in the center.

**Review by fred3m — APPROVED** [review]

**Pull request merged**